shop/src/main/java/ro/msg/learning/shop/controller/ProductController.java

```diff
@@ -0,0 +1,57 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.web.bind.annotation.*;
+import ro.msg.learning.shop.DTO.ProductCategoryDTO;
+import ro.msg.learning.shop.service.ProductService;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+
+    public ProductController(ProductService productService) {
+        this.productService = productService;
+    }
+
+//    @GetMapping("/get")
+//    public List<Product> getAllProducts(){
+//        return productService.findAllProducts();
+//    }
+
+    @GetMapping("/get")
+    public List<ProductCategoryDTO> getAllProducts(){
+        return productService.getAllProducts()
+                .stream()
+                .map(this.productService::mapToDTO)
+                .collect(Collectors.toList());
+    }
+
+    @GetMapping("/product/{id}")
+    public Optional<ProductCategoryDTO> getProductById(@PathVariable(value = "id") Integer id){
```

Comment:
If the method parameter and the PathVariable have the same name then it's not necessary to set the value of the PathVariable specifically. 
Also here, we should not return an Optional. Verify if the object is present in the service layer (business logic).